service/monitoring_service.go

```diff
@@ -17,14 +22,33 @@ type monitoringService struct {
 
 type MonitoringService interface {
 	ProcessEvent(context.Context, models.JobEvent, models.NamespaceSpec, models.JobSpec) error
+	GetJobRunByScheduledAt(context.Context, models.NamespaceSpec, models.JobSpec, time.Time) (models.JobRunSpec, error)
+	GetJobRunByRunID(context.Context, uuid.UUID) (models.JobRunSpec, error)
+}
+
+func (m monitoringService) GetJobRunByScheduledAt(ctx context.Context, namespaceSpec models.NamespaceSpec, jobSpec models.JobSpec, scheduledAt time.Time) (models.JobRunSpec, error) {
+	return m.JobRunMetricsRepository.GetActiveJobRun(ctx, scheduledAt.Format(store.ISODateFormat), namespaceSpec, jobSpec)
```

Comment:
it is not just active job run it should fetch even for old job runs so the method name contradicts.